### ISSUE TITLE
Merge jp's Portamento branch

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/dexed#7]
@jpcima has a portamento branch for dexed hanging out there

Its pretty easy to merge. One spot where I need to rework it to work with tuning infrastructure, but I should do this and we should give it a whirl. @mkruselj volunteered to test if I do

Here's his branch

https://github.com/asb2m10/dexed/pull/183

@jpcima we've done a fork of dexed over here to add some micro tuning and MPE features. You'd be OK with me merging your branch into our fork? If not I will hold off.

**Comment by jpcima** [comment]
Hi @baconpaul, glad to hear you are working to improve Dexed.

I'm not highly reactive in the last days, sorry about that, but I can summarize you the status on portamento work.

The current PR, while it's useful, it's not really matching to DX7 measurements.
@farvardin has been kind enough to provide me some test audio and do some checks, and I have not "cracked the code" since, on how to reproduce it precisely.

[Test DX7 + portamento.mp3.gz](https://github.com/surge-synthesizer/dexed/files/4217155/Test.DX7.%2B.portamento.mp3.gz)
https://github.com/asb2m10/dexed/pull/183#issuecomment-558140172

I made it respond to portamento CC, which I don't know if actual DX7 does, but I guess nice to have.

In DX7, all portamento controls are global settings edited on the instrument.
One of these is "chromatic" mode, which probably should receive a setting in the options panel.
As I remember, it was not complicated to implement it, it's a matter of rounding the porta pitch to whole semitones as it glides.

It's a WIP I lost since a disk crash, but not difficult; and I hadn't completed the control or UI code of it.

**Comment by mkruselj** [comment]
Would be brilliant if all these portamento/glissando options were optionally also stored per patch.

**Comment by jpcima** [comment]
In fact there is some good news, code which I thought to be lost exists online in a branch.

There is glissando and a support for playing with glide in Mono mode.

https://github.com/jpcima/dexed/commits/portamento-wip

**Comment by jpcima** [comment]
However, disregard this following line, which is mistaken.

https://github.com/jpcima/dexed/commit/700e8c2d5b001e064ecc3abf705e67d9cbbbc3ab
`    porta_gliss_ = patch[68];`

The sysex byte 68 is not glissando, this was my misinterpretation of the parameters document, so update accordingly. Portamento stuff are really global params.

**Comment by mkruselj** [comment]
Yeah on the original they are global, but IMHO there should be an option to store individual per-patch settings if we want to. :)

**Comment by baconpaul** [comment]
Yeah @jpcima that wip was the one i merged before - but have to do a bit of merge work since it conflicts lightly with my tuning code. Will do and will let you know
